src/sync_template.sh

```diff
@@ -243,11 +251,15 @@ function pull_source_changes() {
 
   eval "git pull ${source_repo} --tags ${git_remote_pull_params}" || pull_has_issues=true
 
+  gh auth logout --hostname "${SOURCE_REPO_HOSTNAME}"
```

Comment:
I think we need to logout from the source repo. There is the case of same hostname for source and target